compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConstDeclarationChecker.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.lexer.KtTokens
+
+object FirConstDeclarationChecker : FirBasicDeclarationChecker() {
+    override fun check(declaration: FirDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (declaration !is FirVariable<*>) return
+        val source = declaration.source ?: return
+        if (source.kind is FirFakeSourceElementKind) return
+        val modifierList = with(FirModifierList) { source.getModifierList() }
+        val constModifier = modifierList?.modifiers?.firstOrNull { it.token == KtTokens.CONST_KEYWORD } ?: return
```

Comment:
As shown here: https://github.com/JetBrains/kotlin/commit/d7a6fc80b0759200323908e2c19bd819c42ab92d (and mentioned at https://github.com/JetBrains/kotlin/pull/4142#discussion_r587158618, this could be simply `if (!declaration.isConst) return`

Reply:
Actually we do need the modifier anyway to report on, so we might as well keep it this way.

Reply:
Oops, didn't see the use of `constModifier.source`. Seems okay, sorry.